### PR TITLE
Support BF16 in grid_sample

### PR DESCRIPTION
**Pull request by cchan**
Just like https://github.com/pytorch/pytorch/pull/91511, not sure why this was omitted in the first place.

### CODE REVIEW

**Comment by Chillee** [comment]
It looks like there are some errors that look kinda related? I'm not sure why they happen though.

**Comment by ngimel** [comment]
`b16` has to be added to expected grid_sampler xfails in test_meta.py

File: test/test_meta.py [hunk 1]
```diff
@@ -936,7 +936,7 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
     aten._use_cudnn_ctc_loss.Tensor: {f32, f64},  # aten::_use_cudnn_ctc_loss.Tensor
     aten.cudnn_grid_sampler.default: {f16, f32, f64},  # aten::cudnn_grid_sampler
```
Comment (cchan) [inline test/test_meta.py:937]: @ngimel do cudnnSpatialTfSamplerForward and cudnnSpatialTfSamplerBackward support BFloat16?

Comment (ngimel) [reply test/test_meta.py:937]: I don't know, you can try

**Pull request closed without merging**